### PR TITLE
feat: support `as` to change the rendered component

### PR DESCRIPTION
**Pull request by maartenvanhunsel**
Resolves: https://github.com/Barbapapazes/nuxt-authorization/issues/31

### CODE REVIEW

File: src/runtime/components/Tag.ts
Comment (Barbapapazes) [inline src/runtime/components/Tag.ts:1]: can you not use this component instead?

https://vuejs.org/api/built-in-special-elements#component

**Review by Barbapapazes — CHANGES_REQUESTED** [review]

File: src/runtime/components/Tag.ts
Comment (maartenvanhunsel) [reply src/runtime/components/Tag.ts:1]: It's quite the same, however I just wanted to give the user the opportunity to use HTML tags optionally.

With Component :is we cannot create an empty <template> and we always have to render a div or another tag

Comment (maartenvanhunsel) [reply src/runtime/components/Tag.ts:1]: @Barbapapazes so WYT?

Comment (Barbapapazes) [reply src/runtime/components/Tag.ts:1]: Hey! I'll will have a look!

**Review by Barbapapazes — APPROVED** [review]

**Pull request merged**